backend/src/db/courses.js

```diff
@@ -182,20 +179,30 @@ const reduceCoursesByDate = async (courses) => {
                     course_type_id: course.course_type_id,
                     course_type_name: course.course_type_name,
                     teacher: course.teacher,
-                    location: course.location[0].dataValues.location,
-                    address: course.location[0].dataValues.address,
-                    eventId: course.teachingSession[0].dataValues.eventId,
-                    startDate: course.teachingSession[0].dataValues.startDate,
-                    endDate: course.teachingSession[0].dataValues.endDate,
-                    price: price,
+                    location: course.location[0] ? course.location[0].dataValues.location : null,
```

Comment:
you can use lodash `get` next time to do this sort of stuff. It will just return undefined instead of throwing error.
Lodash is safe to use in backend code, since bundle size does not matter too much